### PR TITLE
Style Encoder

### PR DESCRIPTION
**Pull request by deepglugs**
Just experimental for now, but I have some results that are interesting.  The "style" encoder is a pretrained vgg19 network that is fine-tuned with Imagen and concatenated with the text embeddings.  The use-case is to have greater control over the resulting image but can be used with cond_image and/or init_image for ultimate control over the end product.  I will be conducting experiments with different use-cases along the way including style + pose estimate (as cond_image) and style + init_image variations.  Some initial results:

**All Experiments:**
text conditioning prompt: "1girl, blonde, blonde_hair"

### Experiment 1:
Style image

cond_scale=2.0

image

### Experiment 2:
Same style as above with init_image and sigma_max=0.75, cond_scale=2.0
init_image:

Results:
image

### Experiment 3:
Style plus conditioned on pose estimation images. cond_scale=2.0

Row 3 is the OG/style image.

imagen_6_8805_loss0 059061

The model used the pretrained style from above and started producing images conditioned on the pose before the second epoch.  The above was generated during epoch 6.

### Experiment 4:
Use vgg19 pretrained on imagenet.  No additional training

WIP

### CODE REVIEW

**Comment by deepglugs** [comment]
One interesting thing that may be the result of still early training is that high cond_scales result in very bizarre images.  Here's cond_scale=7:
image

**Comment by deepglugs** [comment]
**Update**: After about 16 epochs (on a 30k deepfashon dataset), samples started getting visibly worse.  At 30 epochs, they looked like this:

imagen_30_950_loss0 063138

Unsure why the samples degraded even though loss appeared to still be going down.  I'll go for another 10 epochs to see if things turn around.

**Comment by deepglugs** [comment]
Things got better with my training.  Was probably the conditioning that was affecting the images.  I'll update the original post with results

**Comment by deepglugs** [comment]
update: Added pose conditioned style image initial experimental results.  Because this model used the pretrained style model, it started producing results really fast.  It's clear, however that more training is required.  I can use higher cond_scales, but anything above 5.0 produce really bad results.

**Comment by deepglugs** [comment]
Abandoning this.  It was an interesting experiment, but I don't think it buys anything that text could not achieve.
